test/unit/transform/ferc1_test.py

```diff
@@ -415,6 +420,65 @@ def test_unstack_balances_to_report_year_instant_xbrl():
         )
 
 
+def test_add_columns_with_uniform_value():
+    """Test :func:`add_columns_with_uniform_value`."""
+    df = pd.DataFrame(index=[0, 1, 2])
+    params = AddColumsWithUniformValue(
+        columns_to_add={
+            "utility_type": {"column_value": "electric"},
+            "plant_status": {"column_value": "in_service"},
+        }
+    )
+    df_expected = pd.DataFrame(
+        {"utility_type": ["electric"] * 3, "plant_status": ["in_service"] * 3}
+    )
+    df_out = add_columns_with_uniform_value(df, params)
+    pd.testing.assert_frame_equal(df_expected, df_out)
+
+
+def test_dimension_columns():
+    """Test the :meth:`Ferc1TableTransformParams.dimension_columns`.
+
+    Can ``dimension_columns`` grab a column from :class:`AddColumsWithUniformValue` and ignore
+    a column labled as ``is_dimension=False``?
+
+    Can ``dimension_columns`` also grab a different ``subtotal_column``
+    from :class:`ReconcileTableCalculations`?
+
+    Will ``dimension_columns`` return only one column if the dimension column from
+    :class:`AddColumsWithUniformValue` and :class:`ReconcileTableCalculations` are the same?
+    """
+    add_columns_with_uniform_value = AddColumsWithUniformValue(
+        columns_to_add={
+            "added_dim": AddColumnWithUniformValue(
+                column_value="i'm dim", is_dimension=True
+            ),
+            "not_a_dim": AddColumnWithUniformValue(
```

Comment:
Do you want to assert that this column got added to the dataframe, but not as a dimension column?

Reply:
in here i'm testing the `dimension_columns` parameter class but not i'm not actually doing the transform at all.

in the `test_add_columns_with_uniform_value` above, we're actually doing the transform. in that context i didn't set the `is_dimension` bool so by default they are both `False` and are being added. I could make that explicit above and test it up there or add a new test that does that mixed `is_dimension` transform.

Reply:
okay i added a second step in `test_add_columns_with_uniform_value`